tracer/src/Datadog.Trace/OTelMetrics/MetricStreamIdentity.cs

```diff
@@ -21,6 +23,14 @@ public MetricStreamIdentity(Instrument instrument, InstrumentType instrumentType
         {
             InstrumentName = instrument.Name;
             MeterName = instrument.Meter.Name;
+            MeterVersion = instrument.Meter.Version ?? string.Empty;
+
+#if NET8_0_OR_GREATER
+            MeterTags = instrument.Meter.Tags?.ToArray() ?? [];
+#else
+            MeterTags = [];
+#endif
+
```

Comment:
Hmm I think since we build against anything higher than .NET 6 so this always hits the `else` per my understanding